data/model/user_timeline_event.py

```diff
@@ -81,3 +81,12 @@ class APITimelineEvent(pydantic.BaseModel):
     user_name: str
     created: int
     metadata: APIEventMetadata
+
+class APIRecommendationNotificationTimelineEvent(pydantic.BaseModel):
+    # since `id` is not sent in APITimelineEvent
```

Comment:
You can instead add the `id` field to APITimelineEvent. If all entities are not going to have it, you can make it as `id: Optional[int]`.